test/Microsoft.ML.Core.Tests/UnitTests/TestResourceDownload.cs

```diff
@@ -0,0 +1,276 @@
+#endif
+
+                Assert.True(Uri.TryCreate(path, UriKind.Absolute, out var baseDirUri));
+                Environment.SetEnvironmentVariable(ResourceManagerUtils.CustomResourcesUrlEnvVariable, baseDirUri.AbsoluteUri);
+                envVar = Environment.GetEnvironmentVariable(ResourceManagerUtils.CustomResourcesUrlEnvVariable);
+                if (envVar != baseDirUri.AbsoluteUri)
+                    Fail("Environment variable not set properly");
+
+                Environment.SetEnvironmentVariable(ResourceManagerUtils.CustomResourcesUrlEnvVariable, null);
+                envVar = Environment.GetEnvironmentVariable(ResourceManagerUtils.CustomResourcesUrlEnvVariable);
+                if (envVar != null)
+                    Fail("Environment variable not set properly");
+
+                Environment.SetEnvironmentVariable(ResourceManagerUtils.TimeoutEnvVariable, "10");
+                envVar = Environment.GetEnvironmentVariable(ResourceManagerUtils.TimeoutEnvVariable);
+                if (envVar != "10")
+                    Fail("Environment variable not set properly");
+
+                DeleteOutputPath("copyto", "ResNet_18_Updated.model");
+                sbOut.Clear();
+                sbErr.Clear();
+                using (var outWriter = new StringWriter(sbOut))
+                using (var errWriter = new StringWriter(sbErr))
+                {
+                    var env = new ConsoleEnvironment(42, outWriter: outWriter, errWriter: errWriter);
+                    using (var ch = env.Start("Downloading"))
+                    {
+                        var fileName = "test_short_timeout.model";
+                        var t = ResourceManagerUtils.Instance.EnsureResourceAsync(env, ch, "Image/AlexNet_Updated.model", fileName, saveToDir, 10 * 1000);
+                        t.Wait();
+
+                        Log("Default url, short time out");
+                        Log($"out: {sbOut.ToString()}");
+                        Log($"error: {sbErr.ToString()}");
+
+#if !CORECLR
+                        if (!sbErr.ToString().Contains("Download timed out"))
+                            Fail($"Error message should contain the string 'Download timed out'. Instead: {sbErr.ToString()}");
+#endif
+                        if (File.Exists(Path.Combine(saveToDir, fileName)))
+                            Fail($"File '{Path.Combine(saveToDir, fileName)}' should have been deleted.");
+                    }
+                }
+                Done();
+            }
+            finally
+            {
+                // Set environment variable back to its old value.
+                Environment.SetEnvironmentVariable(ResourceManagerUtils.CustomResourcesUrlEnvVariable, envVarOld);
+                Environment.SetEnvironmentVariable(ResourceManagerUtils.TimeoutEnvVariable, timeoutVarOld);
+                Environment.SetEnvironmentVariable(Utils.CustomSearchDirEnvVariable, resourcePathVarOld);
+            }
+        }
```

Comment:
Specifically testing the code path which detects bad aka.ms would be helpful to ensure our detection logic remains working.

Potential URI:  `http://aka.ms/mlnet/badurltest/{Guid.NewGuid()}`. The added "mlnet/badurltest" token is to be nice to the aka.ms folks who would otherwise receive unexplainable 404s.

Reply:
Good point, I've replaced the earlier URL test with this to avoid unnecessary extra test cases and longer testing times.